torchgeo/datasets/dior.py

```diff
@@ -0,0 +1,390 @@
+            int(bndbox.find('xmin').text),  # type: ignore[union-attr, arg-type]
+            int(bndbox.find('ymin').text),  # type: ignore[union-attr, arg-type]
+            int(bndbox.find('xmax').text),  # type: ignore[union-attr, arg-type]
+            int(bndbox.find('ymax').text),  # type: ignore[union-attr, arg-type]
+        ]
+        label = obj.find('name').text  # type: ignore[union-attr]
+        bboxes.append(bbox)
+        labels.append(label)
+
+    return dict(filename=filename, bboxes=bboxes, labels=labels)
+
+
+class DIOR(NonGeoDataset):
+    """DIOR dataset.
+
+    `DIOR <https://arxiv.org/abs/1909.00133>`_ dataset contains horizontal bounding box
+    annotations of Google Earth Aerial RGB imagery. The test split does not contain bounding
+    box annotations and labels
+
+    Dataset features:
+
+    * 20 classes
+    * 192,472 manually annotated bounding box instances
+
+    Dataset format:
+
+    * Images are three channel .jpg files.
+    * Annotations are in `Pascal VOC XML format
+      <https://roboflow.com/formats/pascal-voc-xml>`_
+
+
+    Classes:
+
+    0. Airplane
+    1. Airport
+    2. Baseball Field
+    3. Basketball Court
+    4. Bridge
+    5. Chimney
+    6. Dam
+    7. Expressway Service Area
+    8. Expressway Toll Station
+    9. Golf Field
+    10. Ground Track Field
+    11. Harbor
+    12. Overpass
+    13. Ship
+    14. Stadium
+    15. Storage Tank
+    16. Tennis Court
+    17. Train Station
+    18. Vehicle
+    19. Windmill
+
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/1909.00133
+
+
+    .. versionadded:: 0.7
+    """
+
+    url = 'https://hf.co/datasets/torchgeo/dior/resolve/ec7be9567d2e08eb3d3401c15a52ee2145d0ef01/{}'
+
+    files: ClassVar[dict[str, dict[str, dict[str, str]]]] = {
+        'trainval': {
+            'images': {
+                'filename': 'Images_trainval.zip',
+                'md5': '070e9314120403e5c965d12fe5321cb0',
+            },
+            'labels': {
+                'filename': 'Annotations_trainval.zip',
+                'md5': '90e045de37255c5919bbecf659b72c1a',
+            },
+        },
+        'test': {
+            'images': {
+                'filename': 'Images_test.zip',
+                'md5': '97f3cbc86de0867624a6a34190c694ae',
+            }
+        },
+    }
+
+    valid_splits = ('train', 'val', 'test')
+
+    classes = (
+        'airplane',
+        'airport',
+        'baseballfield',
+        'basketballcourt',
+        'bridge',
+        'chimney',
+        'dam',
+        'expresswayservicearea',
+        'expresswaytollstation',
+        'golffield',
+        'groundtrackfield',
+        'harbor',
+        'overpass',
+        'ship',
+        'stadium',
+        'storagetank',
+        'tenniscourt',
+        'trainstation',
+        'vehicle',
+        'windmill',
+    )
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
```

Comment:
```suggestion
        split: Literal['train', 'val', 'test'] = 'train',
```
Trying to push use of `typing.Literal` in more cases